tests/test_repr.py

```diff
@@ -24,3 +24,14 @@ def test_str_spawn(self):
         # verify
         assert isinstance(value, str)
 
+    def test_str_before_spawn(self):
+        """ Exercise derived spawn.__str__() """
+        # given,
+        child = pexpect.spawn(None, None)
+        child.read_nonblocking = lambda size, timeout: b''
+        try:
+            child.expect('alpha', timeout=0.1)
+        except pexpect.TIMEOUT:
```

Comment:
Actually, I don't think this test case would exhibit the failure that you mentioned, because it arises when something tries to *display* the error, and if you catch it, nothing will do that.